tests/sharded_jit_test.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from functools import partial
+from unittest import SkipTest
+
+import numpy as onp
+from absl.testing import absltest
+
+import jax
+import jax.numpy as np
+from jax import test_util as jtu
+from jax.interpreters import pxla
+from jax.interpreters.sharded_jit import sharded_jit
+from jax.interpreters.sharded_jit import PartitionSpec as P
+
+from jax.config import config
+config.parse_flags_with_absl()
+
+
+class ShardedJitTest(jtu.JaxTestCase):
+
+  def testBasic(self):
+    if jtu.device_under_test() != "tpu" or jax.device_count() < 2:
+      raise SkipTest
+
+    @partial(sharded_jit, in_parts=(P(2, 1), P(2, 1)), out_parts=None)
+    def f(x, y):
+      return x + y
+
+    shape = (8, 8)
+    x = onp.arange(onp.prod(shape), dtype=np.float32).reshape(shape)
+    actual = f(x, x + 1)
+    expected = x + (x + 1)
+    self.assertAllClose(actual, expected, check_dtypes=False)
+    self.assertIsInstance(actual, pxla.ShardedDeviceArray)
+    self.assertLen(actual.device_buffers, 2)
+    self.assertAllClose(actual.device_buffers[0].to_py(), expected,
+                        check_dtypes=False)
+
+  def testPyTreeArgs(self):
+    if jtu.device_under_test() != "tpu" or jax.device_count() < 2:
+      raise SkipTest
+
+    def f(a, b, c):
+      a1, a2 = a
+      c1, (c2, c3) = c
+      return a1 + a2 + b + c1 + c2 + c3
+
+    def _make_arg(*shape):
+      return onp.arange(onp.prod(shape)).reshape(shape)
+
+    a = (_make_arg(4, 4), _make_arg())
+    b = _make_arg(4, 4)
+    c = [_make_arg(), (_make_arg(4, 4), _make_arg(4, 4))]
```

Comment:
Is `make_arg` producing a 0?

Reply:
Yes, `np.prod(())` returns 1.0 (so this is actually 0.).

Reply:
Then wouldn't it be easier to see what is going on by using 0 instead of `make_arg()`

Reply:
Yup, done! (I made them non-zero ints for good measure.)